crates/uv-python/src/downloads.rs

```diff
@@ -111,6 +111,33 @@ pub enum Error {
     },
 }
 
+impl Error {
+    // Return the number of attempts that were made to complete this request before this error was
+    // returned. Note that e.g. 3 retries equates to 4 attempts.
+    //
+    // It's easier to do arithmetic with "attempts" instead of "retries", because if you have
+    // nested retry loops you can just add up all the attempts directly, while adding up the
+    // retries requires +1/-1 adjustments.
+    fn attempts(&self) -> u32 {
+        // Unfortunately different variants of `Error` track retry counts in different ways. We
+        // could consider unifying the variants we handle here in `Error::from_reqwest_middleware`
+        // instead, but both approaches will be fragile as new variants get added over time.
```

Comment:
The fragility here makes me feel like I'm doing something wrong. @konstin do you have any better ideas?

Reply:
iirc streaming error can show up as different error types (e.g. `Error::ExtractError`), but we're only attaching the retries if we have a reqwest error